Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithVariableSpeed.mo

```diff
@@ -0,0 +1,244 @@
+  parameter Modelica.Units.SI.Efficiency epsSenHeaPL(
+    final max=1) = 0.75
+    "Part load (75%) sensible heat exchanger effectiveness at the heating mode";
+  parameter Modelica.Units.SI.Efficiency epsLatHeaPL(
+    final max=1) = 0.75
+    "Part load (75%) latent heat exchanger effectiveness at the heating mode";
+
+  Modelica.Blocks.Interfaces.RealInput uSpe(
+    final unit="1",
+    final max=1)
+    "Wheel speed"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+  Modelica.Blocks.Interfaces.RealOutput P
+    "Electric power consumed by the wheel"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+  Buildings.Fluid.HeatExchangers.AirToAirHeatRecovery.BaseClasses.HeatExchangerWithInputEffectiveness
+    hex(
+    redeclare package Medium1 = Medium1,
+    redeclare package Medium2 = Medium2,
+    final m1_flow_nominal=m1_flow_nominal,
+    final m2_flow_nominal=m2_flow_nominal,
+    final show_T=true,
+    final dp1_nominal=dp1_nominal,
+    final dp2_nominal=dp2_nominal)
+    "Heat exchanger"
+    annotation (Placement(transformation(extent={{6,-10},{26,10}})));
+  Buildings.Fluid.HeatExchangers.AirToAirHeatRecovery.BaseClasses.Effectiveness
+    effCal(
+    final epsSenCoo_nominal=epsSenCoo_nominal,
+    final epsLatCoo_nominal=epsLatCoo_nominal,
+    final epsSenCooPL=epsSenCooPL,
+    final epsLatCooPL=epsLatCooPL,
+    final epsSenHea_nominal=epsSenHea_nominal,
+    final epsLatHea_nominal=epsLatHea_nominal,
+    final epsSenHeaPL=epsSenHeaPL,
+    final epsLatHeaPL=epsLatHeaPL,
+    final VSup_flow_nominal=m1_flow_nominal/1.293)
+    "Calculate the effectiveness of heat exchange"
+    annotation (Placement(transformation(extent={{-40,40},{-20,60}})));
+  Buildings.Fluid.Sensors.VolumeFlowRate senExhFlow(
+    redeclare package Medium = Medium2,
+    final m_flow_nominal=m2_flow_nominal)
+    "Exhaust air volume flow rate sensor"
+    annotation (Placement(transformation(extent={{70,-70},{50,-50}})));
+  Buildings.Fluid.Sensors.VolumeFlowRate senSupFlow(
+    redeclare package Medium =Medium1,
+    final m_flow_nominal=m1_flow_nominal)
+    "Supply air volume flow rate sensor"
+    annotation (Placement(transformation(extent={{-40,90},{-20,110}})));
+  Modelica.Blocks.Sources.RealExpression TSup(
+    final y=Medium1.temperature(
+      Medium1.setState_phX(
+        p=port_a1.p,
+        h=inStream(port_a1.h_outflow),
+        X=inStream(port_a1.Xi_outflow))))
+    "Supply air temperature"
+    annotation (Placement(transformation(extent={{-90,28},{-70,48}})));
+  Modelica.Blocks.Sources.RealExpression TExh(
+    final y=Medium2.temperature(
+      Medium2.setState_phX(
+        p=port_a2.p,
+        h=inStream(port_a2.h_outflow),
+        X=inStream(port_a2.Xi_outflow))))
+    "Exhaust air temperature"
+    annotation (Placement(transformation(extent={{-90,10},{-70,30}})));
+  Modelica.Blocks.Sources.RealExpression PEle(
+    final y=P_nominal)
```

Comment:
Is it correct? With this calculation, the power consumption will always equal to the nominal value.